### DISCUSSION TITLE
Getting "Could not retrieve database metadata" for Oracle DB Reader

### DISCUSSION THREAD

**Discussion by LanciaH** [nextgenhealthcare/connect#5169] [Q&A] [answered]
Hi All,

I trying to connect to a Oracle database using oracle.jdbc.driver.OracleDriver, "Use JavaScript" off  No matter what I try I get the following error; Could not retrieve database metadata.  Please ensure that your driver, URL, username, and password are correct.
Method failed: HTTP/1.1 500 Internal Server Error

I have tried different Oracle databases and they all give the same error.  On the same PC I can connect via java programs and TOAD, TOAD uses TNSNAMEs but my JAVA apps use oracle.jdbc.driver.OracleDriver with the ojdbc8.jar file.  Any thoughts would be helpful.
Thanks,
Ralph

**Comment by LanciaH** [comment]
I copied the database tnsnames entry in the URl and my select works, but when I click on Select --> GET TABLES is when the error occurs.  Its okay I can live with this.

Thanks

**Reply by pacmano1** [reply]
If you put an asterisk in the search box does it work?   I think it always fails otherwise.

**Reply by pacmano1** [reply]
Perhaps related: https://github.com/nextgenhealthcare/connect/issues/4876

**Reply by LanciaH** [reply]
Actually nothing happens.

Thanks for the suggestion

**Comment by jonbartels** [comment]
The relevant code that does this is in `com.mirth.connect.connectors.jdbc.DatabaseConnectorServlet#getTables` . You can even call that from the API if you wanted to. Under the covers it calls bog standard JDBC methods, `java.sql.Connection#getMetaData`. Any Java based DB tool should have similar behaviors - SquirrelSQL, DataGrip.

Are you using any mapped values for your connection string or credentials? If you are those will work at runtime but will not generally work when testing connections (which is effectively what `getTables` does).

What errors, if any, are in `mirth.log`? The HTTP 500 you see won't have much detail but the server logs for MC should.

**Comment by LanciaH** [comment, marked as answer]
Never could getTables to work.  The SQL query works fine and the data gets converted into XML.
Much better than working with files.
